src/Map.js

```diff
@@ -66,6 +66,14 @@ export class Map extends KeyedCollection {
     });
   }
 
+  static ownEntries(obj) {
+    if (typeof obj !== 'object') {
+      return obj;
+    }
+
+    return Map(Reflect.ownKeys(obj).map(k => [k, Map.ownEntries(obj[k])]));
```

Comment:
I don't think I understand why this is recursing, that seems incorrect

Reply:
i believe i have at least one test case/assertion which requires recursion. perhaps that case is invalid, or unclear. i'll start there and add clarity to the test case. if it's invalid, we'll drop it : )